src/wazuh_db/wdb_agents.c

```diff
@@ -94,21 +94,37 @@ cJSON* wdb_agents_insert_vuln_cves(wdb_t *wdb,
     return result;
 }
 
-int wdb_agents_update_status_vuln_cves(wdb_t *wdb, const char* old_status, const char* new_status) {
+int wdb_agents_update_status_vuln_cves(wdb_t *wdb, const char* old_status, const char* new_status, const char* type) {
+    sqlite3_stmt* stmt = NULL;
 
-    bool update_all = (strcmp(old_status, "*") == 0);
+    if (old_status && new_status && !type) {
```

Comment:
I suggest removing the last part of the condition `&& !type`. In this block of code within the `if` statement you will use only `old_status` and `new_status`. The `type` will not be used at all so it wouldn't be verified.

Reply:
I'd rather leave the extra check, just to be sure the query isn't called with three parameters and generate an unexpected behavior.

Reply:
Good! So let's keep it.